weave/Scripts/CurveSpawner.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using Godot;
+using GodotSharper.Instancing;
+using weave;
+
+[Instantiable("res://Objects/CurveSpawner.tscn")]
+public partial class CurveSpawner : Node2D
+{
+    public Player Player { get; set; }
+    public List<SegmentShape2D> Segments { get; set; } = new();
+    public int LineWidth { get; set; }
+    private readonly float curveSpawnOffset = 4f; 
+    private Vector2 lastPoint;
+    private bool hasStarted = false;
+    private Color lineColor = new(1, 0, 0);
+
+    public override void _Process(double delta)
+    {
+        if (Player != null)
+        {
+            var playerShape = (CircleShape2D)Player.collisionShape2D.Shape;
```

Comment:
We should probably cache this shape. (and use guard clause for if player is null instead?)